### PR TITLE
added event types and some minor changes

### PR DESCRIPTION
**Pull request by himanshu-spr**

### CODE REVIEW

File: src/components/AddEditForm/index.tsx [hunk 1]
```diff
@@ -20,41 +20,50 @@ const yearOptions = DropdownFilterConstant.year.options
...
+      const curCourse = {
+        name: courseState.name,
+        prof: "John Doe",
+        strength: props.course.strength,
+        branch: courseState.branch,
+        year: courseState.year,
+        code: courseState.code,
+        credits: courseState.credits,
+      };
+      props.onSubmit(curCourse);
```
Comment (NiketMalik) [inline src/components/AddEditForm/index.tsx]:
```suggestion
      props.onSubmit({ ...courseState, prof: "John Doe" });
```

Comment (NiketMalik) [reply src/components/AddEditForm/index.tsx]: Should also check if props.onSubmit is not undefined

File: src/components/AddEditForm/index.tsx [hunk 2]
```diff
@@ -20,41 +20,50 @@ const yearOptions = DropdownFilterConstant.year.options
...
+    },
+    [courseState]
```
Comment (NiketMalik) [inline src/components/AddEditForm/index.tsx]: props.onSubmit will also be a dependency

File: src/components/CourseView/CourseGridView/index.tsx [hunk 3]
```diff
@@ -1,14 +1,18 @@
...
+import React, { useCallback } from "react";
 import "./CourseGridView.css";
 import { CourseGridViewProps } from "../../../interfaces";
 import GridCourse from "./GridCourse";
 
 const CourseGridView = (props: CourseGridViewProps) => {
-  const coursesElement = props.courses.map((course) => {
-    return <GridCourse course={course} />;
-  });
+  const coursesElement = useCallback(
```
Comment (NiketMalik) [inline src/components/CourseView/CourseGridView/index.tsx]: renderCourses

File: src/components/CourseView/CourseListView/index.tsx [hunk 4]
```diff
@@ -1,12 +1,16 @@
...
+import React, { useCallback } from "react";
 import "./CourseListView.css";
 import { CourseListViewProps } from "../../../interfaces";
 import ListCourse from "./ListCourse";
 
 const CourseListView = (props: CourseListViewProps) => {
-  const coursesElement = props.courses.map((course) => {
-    return <ListCourse course={course} />;
-  });
+  const coursesElement = useCallback(
```
Comment (NiketMalik) [inline src/components/CourseView/CourseListView/index.tsx]: Extract this to a common helper

File: src/components/SearchFilter/index.tsx [hunk 5]
```diff
@@ -1,31 +1,31 @@
...
 
 const SearchFilter = () => {
-    const [state, dispatch] = useContext(FilterContext);
+  const [state, dispatch] = useContext(FilterContext);
 
-    const changeHandler = (event: any) => {
-        event.preventDefault();
-        dispatch({type: 'MODIFY_SEARCH', payload: event.target.value});
-    };
+  const changeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (NiketMalik) [inline src/components/SearchFilter/index.tsx]: should be in useCallback

File: src/components/AddCourse/index.tsx [hunk 6]
```diff
@@ -18,6 +18,7 @@ const AddCourse = () => {
 
   const submitFormHandler = useCallback((course) => {
     dispatch({ type: "ADD", payload: course });
+    closeFormHandler();
```
Comment (NiketMalik) [inline src/components/AddCourse/index.tsx]: closeFormHandler will be in dependency also

**Review by NiketMalik — CHANGES_REQUESTED** [review]

File: src/components/AddEditForm/index.tsx (hunk 1 above)
Comment (himanshu-spr) [reply src/components/AddEditForm/index.tsx]: I did not understand why do we need to check this.

File: src/components/AddEditForm/index.tsx [hunk 7]
```diff
@@ -20,41 +20,45 @@ const yearOptions = DropdownFilterConstant.year.options
...
-
-  const handleOnChange = useCallback((event: any) => {
-    let value = event.target.value;
-    if (event.target.name === "year" || event.target.name === "credits") {
-      value = parseInt(event.target.value);
-    }
+  const submitHandler = useCallback(
+    (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      props.onSubmit({
```
Comment (NiketMalik) [inline src/components/AddEditForm/index.tsx]: SHould check if `props.onSubmit` is undefined or not, always make it a habit of checking functions from prop

**Review by NiketMalik — CHANGES_REQUESTED** [review]

**Pull request merged**